docs/proposals/multi-node-cluster.md

```diff
@@ -0,0 +1,212 @@
+  `--all` starts all nodes already created.
+
+* `stop <worker_name>`
+
+  Stops a node keeping its state, does not remove it from cluster's API
+
+* `delete <worker_name>`
+
+  Deletes a node, attempting to delete it from the API if the master is running.
+
+* `ssh <worker_name> -- [command]`
+
+  Opens a shell on the node, optionally runs a command.
+
+* `docker-env <worker_name>`
+
+  Prints the docker environment settings for connecting the docker client
+  to this node.
+
+* `status [worker_name]`
+
+  Prints status of specified node, or all nodes if not specified.
+
+* `logs [worker_name]`
+
+  Prints or tails logs of the Kube components running on the node,
+  similar to `minikube logs`.
+
+* `ip <worker_name>`
+
+  Print the IP of the specified node
+
+
+#### Node configuration
+
+Nodes would inherit the configuration of the master node from `minikube start`, but could be customized via flags provided to the start command, allowing one to have different resource capacity on different nodes as well as testing different configurations, including different versions of worker components, eg:
+
+```
+  minikube worker create node-1 \
+    --cpus=1 \
+    --memory=1024 \
+    --disk-size=10g \
+    --extra-opts=... \
+    --container-runtime... \
+    --kubernetes-version=... \
+    --iso-url=... \
+    --docker-env=... \
+    --docker-opt=... \
+    --feature-gates=... \
+    ...
+```
+
+In short, most configuration options that can be passed to `minikube start` should be applicable to a worker node.
+
+### Driver support
+
+_What drivers can support multi-node clusters?_
+
+The primary requirement is that each node must have a unique IP that is routable among all nodes.  Some network plugins require layer 2 connectivity between the nodes, while others encapsulate inter-pod traffic inside IP packets and can operate without layer-2 connectivity between all nodes.  Driver support may impact CNI support if layer 2 networking is not available on all drivers we want to support. We could say that layer-2 is a requirement, so that encapsulation of packets is never needed.
+
+For example, if `flanneld`'s `host-gw` backend will only work if all nodes can see each other via layer-2.  If that is not available, the `vxlan` backend can be used, at a performance and complexity cost (probably not significant at minikube scale).
+
+If a driver supports the above, theoretically it supports a multi-node setup.  An initial implementation could start with `virtualbox` support only, since it is the default driver.
+
+
+### Bootstrapper support
+
+Multi-node will only be supported by the `kubeadm` bootstrapper, as it is designed for multi-node, while `localkube` is not.
+
+
+### Networking
+
+A multi-node cluster relies on a CNI plugin deployed as a DaemonSet to handle routing between nodes.  Minikube installs a default CNI distribution out-of-the-box.  This can be disabled, allowing a user to install their own choice of CNI plugin, eg. `--cni-plugin=none`.
```

Comment:
I think in this case we may need a different flag than `network-plugin`, or a change in the semantics of this flag, because here we're talking about setting which CNI plugin to use.

The `network-plugin` [flag](https://kubernetes.io/docs/concepts/cluster-administration/network-plugins/#usage-summary) just tells localkube (or kubelet) whether to use `cni` or `kubenet`, 
whereas I am proposing we have a flag that tells `minikube` which CNI plugin should be installed on the cluster automatically, eg. flannel, calico, weave.

If we let minikube handle installing the CNI plugin, we would want to force the value of this flag to `cni`, or return an error if the value is not `cni`.